server/src/main/java/org/apache/druid/server/http/DataSourcesResource.java

```diff
@@ -410,7 +414,74 @@ public Response getServedSegmentsInInterval(
       }
       return Response.ok(segmentIds).build();
     }
-    return getServedSegmentsInInterval(dataSourceName, full != null, theInterval::contains);
+    return getServedSegmentsInInterval(dataSource, full != null, theInterval::contains);
+  }
+
+  @GET
+  @Path("/{dataSourceName}/unused/intervals")
+  @Produces(MediaType.APPLICATION_JSON)
+  @ResourceFilters(DatasourceResourceFilter.class)
+  public Response getIntervalsWithUnusedSegmentsOrAllUnusedSegmentsPerIntervals(
+      @PathParam("dataSourceName") String dataSourceName,
+      @QueryParam("simple") String simple,
```

Comment:
Do we need two `String` params for `simple` and `full`? Or would a single `Boolean simple` suffice? (Mostly curious because I see this is an existing pattern, so I don't feel strongly either way)

Reply:
removed